### ISSUE TITLE
Testing for not directory exists not working...

### ISSUE THREAD

**Issue by Panoplos** [fish-shell/fish-shell#4383]
Version: 2.4.0
Environment: Ubuntu 17.04, Konsole (xterm-256color)

I have a function that is testing for the existence of a user-supplied directory with:

```fish
set targetdir $argv[1]
if ! test -d $targetdir
  "The directory you secified ($targetdir) does not exist. Exiting without shooting a duck."
  return -1
end
```
Which is resulting in:

```shell
!: command not found
tools/ducks-cli.fish (line 12):         if ! test -d $targetdir
                                           ^
in function “shoot-a-duck”
        called on standard input
        with parameter list “src/universal/ducks nav setLinks”
```

**Comment by faho** [comment]
@Panoplos: Your script isn't valid in fish.

You'll want to use `not` instead of `!`. I.e.:

```fish
set targetdir $argv[1]
if not test -d $targetdir
  "The directory you specified ($targetdir) does not exist. Exiting without shooting a duck."
  return 1
end
```

(Also, the return status has to be positive - returning `-1` currently results in `255` being returned).

**Comment by Panoplos** [comment]
OK, then the docs should be changed to reflect this, as the current documentation for `test` says to use `!`.

**Comment by faho** [comment]
`test` itself also accepts `!` as an argument, so you could also do `test ! -d` - note that the `!` comes after.

However, in this case, it's clearer to do `not test`.

**Comment by Panoplos** [comment]
Ah, OK. Well, without an actual example of `test !` in its doc, it was not obvious, so I still recommend changing it for clarity. ;^)

**Comment by krader1961** [comment]
@Panoplos, Can you provide an example of how the documentation could be improved? I ask because the `test` documentation seems clear to me that `!` only applies to the evaluated expression, not the `test` command itself. In other words the documentation is explaining that you can do `test ! x = y` rather than `! test x = y`. But I'm a grey beard who is accustomed to this type of subtlety. Now that you understand the difference it would help us if you could provide wording that would have helped you avoid that misunderstanding. If nothing else we should probably include a mention of, and example of using, `not test...` where the `!` operator is mentioned.

**Comment by Panoplos** [comment]
I guess the `! EXPRESSION` reference to `EXPRESSION` at the top of the `test` documentation was lost on a quick glance. Operator error, but I guess the terseness of the docs left room for this.

Maybe changing `! EXPRESSION` to `test ! EXPRESSION` would help the lower IQ among us, like myself?

**Comment by mqudsi** [comment]
> Maybe changing ! EXPRESSION to test ! EXPRESSION would help

I loaded `doc_src/test.txt` up in my editor fully intending to make this change, but then I saw its context, and I can't say I agree at all. The entire document is made up of bullets following the format 

> * `ARGUMENT` _explanation_

The document in question is the documentation for the fish builtin `test`, and not for fish itself. I really hate to say it, but it's exceedingly clear as currently worded. The entire document omits the repeated usage of the `test` command name; which I even considered changing before I realized that all builtins were similarly documented.
